src/monitor.c

```diff
@@ -162,7 +165,7 @@ static void export_stats(int_status_t *intrnl, export_status_t *exp,
 	uint32_t total_tried_sent = iterator_get_tried_sent(it);
 	uint32_t total_fail = iterator_get_fail(it);
 	uint32_t total_recv = zrecv.pcap_recv;
-	uint32_t recv_success = zrecv.success_unique;
+	uint32_t recv_success = zrecv.filter_success;
```

Comment:
Do we lose something weird here because filter_success might not be unique?

Reply:
I think it's a gain. recv_success is only used in monitor to calculate rate and avg. In my opinion, the user should be presented with a rate and average that matches their chosen output filter, not necessarily unique successes.

Reply:
I'm wavering on this. I think we have to decide fundamentally, is filtering _part_ of receiving, or is it _post_ receiving, and we should be consistent with all of the counters.

Reply:
along those lines, it would probably be good to go through all the counters in metadata output, updates file, and on-screen updates, and make sure that we have a good definition of all of these and document exactly what they are, and then double check that this matches up with the code base.

Reply:
Can do 👍 I'll look into those counters, see if I can get more comments/consistency. 

FWIW I think that filtering is already a part of receiving since we're filtering success_unique, and therefore not representing the total number of packets ever received. But I can also see the other side too ¯\_(ツ)_/¯

Reply:
@ajholland did you ever have a chance to look at those values? I just realized that we never actually merged this PR due to that pending question.

Reply:
So here's all the vars: 
total_sent - total number of **hosts** scanned, this pr makes it _packets_ sent
total_tried_sent - count number of IP's that are iterated through in the cyclic group, including those scanned and not scanned. Used to restrict max targets and compute time remaining. this pr makes this counter also increment per packet sent
recv_success_unique - exported var, set to recv_success which in this PR is filter success 
app_recv_success_unique - alias for zrecv.app_success_unique
total_recv - alias for zrecv.pcap_recv, counter of total number pkts, value set from pcap_stat.ps_recv which is number of packets transited on the network 
percent_complete - dependent on age and remaining_secs
remaining_secs - logic in compute_remaining_time function 
send_rate - total sent / delta
recv_rate - recv success / delta
recv_total_rate - total recv / delta
app_success_rate - app_success / delta where app_success is zrecv.app_success_unique
pcap_drop - number of packets dropped by the driver
time_remaining - alias for remaining_secs 
fail_total - packets failed to reach NIC
hitrate - recv_success * 100 / total_sent 
app_hitrate - app_success * 100 / total_sent

as a note, @zakird I think I'm finding more instances where type overflow would occur because of counting packets instead of ip's (all 5 vars above line 168 here). I'll defer to fixing those until you get a chance to comment on whether that is a change you want. 

When I looked through these, the only ones that seem to be inconsistent in terms of definition and code is sent and tried_sent (and that's fixed in the PR). 

Also, thoughts on renaming zconf.packet_streams?

Reply:
Given that this is going to go out with ZMap 3.0, we have the opportunity to change names in this file. I think we should we should do so in order to remove ambiguities. 

Here's what I'm thinking:


`total_sent` - total number of hosts scanned, this pr makes it packets sent *Change: packets_sent. Also add `hosts_scanned`.*
`total_tried_sent` - count number of IP's that are iterated through in the cyclic group, including those scanned and not scanned. Used to restrict max targets and compute time remaining. this pr makes this counter also increment per packet sent *candidate_ips*
`recv_success_unique` - exported var, set to recv_success which in this PR is filter success
app_recv_success_unique - alias for zrecv.app_success_unique *I'm confused about this one is.*
`total_recv` - alias for zrecv.pcap_recv, counter of total number pkts, value set from pcap_stat.ps_recv which is number of packets transited on the network *Change: `response_packets_received`*
percent_complete - dependent on age and remaining_secs
remaining_secs - logic in compute_remaining_time function *-> time_remaining*
send_rate - total sent / delta
recv_rate - recv success / delta *Change: make clear success*
recv_total_rate - total recv / delta *Why does this have a total?*
app_success_rate - app_success / delta where app_success is zrecv.app_success_unique
pcap_drop - number of packets dropped by the driver *Change: pcap_dropped_packets*
time_remaining - alias for remaining_secs *Sure seems like we don't need both of these*
fail_total - packets failed to reach NIC
hitrate - recv_success * 100 / total_sent 
app_hitrate - app_success * 100 / total_sent

Reply:
These are all works in progess, not tied to any but wanted to start to get down

Reply:
I'm confused about how a bunch of these interact with output filters.

Reply:
Note: I'm going to move these to a google spreadsheet and we can discuss there

Reply:
Pinging to see what progress there is on this.